### ISSUE TITLE
False positive with unused type declaration

### ISSUE THREAD

**Issue by coderaiser** [typescript-eslint/typescript-eslint#659]
**What code were you trying to parse?**

```ts
import { Lines } from './lines';
const parts: (string | Lines)[] = [];
```

**What did you expect to happen?**

I expect to see
```
  3:7   error  'parts' is assigned a value but never used  no-unused-vars
```

**What actually happened?**

`Lines` detected as unused, but it is used.
```
 1:10  error  'Lines' is defined but never used           no-unused-vars
 3:7   error  'parts' is assigned a value but never used  no-unused-vars
```

Here is my `.eslintrc.json`:
```json
{
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "sourceType": "module"
    },
    "extends": [
        "eslint:recommended"
    ]
}
```
**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.11.0` |
| `TypeScript`                | `3.4.5` |
| `ESLint`                    | `6.0.1` |
| `node`                      | `12.4.0` |
| `npm`                       | `6.9.0` |

**Comment by bradzacher** [comment]
Duplicate of #363
